pkg/handlers/jsonl_test.go

```diff
@@ -0,0 +1,603 @@
+			t.Errorf("got i.Indexes[0].BPTree().Find(\"test1\") = nil, want non-nil")
+		}
+		if mp1.Offset != 0 || mp1.Length != uint32(len("{\"test\":\"test1\"}")) {
+			t.Errorf("got i.Indexes[0].BPTree().Find(\"test1\") = %+v, want {0, %d}", mp1, len("{\"test\":\"test1\"}"))
+		}
+
+		buf1, err := collected[0].Metadata()
+		if err != nil {
+			t.Fatal(err)
+		}
+		md1 := &appendable.IndexMeta{}
+		if err := md1.UnmarshalBinary(buf1); err != nil {
+			t.Fatal(err)
+		}
+		if md1.FieldType != appendable.FieldTypeString {
+			t.Errorf("got i.Indexes[0].FieldType = %#v, want FieldTypeString", md1.FieldType)
+		}
+
+		mp2, found, err := collected[1].BPTree(r2).Find([]byte("test3"))
+		if err != nil {
+			t.Fatal(err)
+		}
+		if !found {
+			t.Errorf("got i.Indexes[1].BPTree().Find(\"test3\") = nil, want non-nil")
+		}
+		if mp2.Offset != uint64(len("{\"test\":\"test1\"}\n")) || mp2.Length != uint32(len("{\"test2\":\"test3\"}")) {
+			t.Errorf("got i.Indexes[1].BPTree().Find(\"test3\") = %+v, want {%d, %d}", mp2, len("{\"test\":\"test1\"}\n"), len("{\"test2\":\"test3\"}"))
+		}
+
+		md2 := &appendable.IndexMeta{}
+		if err := collected[1].UnmarshalMetadata(md2); err != nil {
+			t.Fatal(err)
+		}
+		if md2.FieldType != appendable.FieldTypeString {
+			t.Errorf("got i.Indexes[1].FieldType = %#v, want FieldTypeString", md2.FieldType)
+		}
+	})
+
+	t.Run("existing index but different type", func(t *testing.T) {
+		f := buftest.NewSeekableBuffer()
+
+		i, err := appendable.NewIndexFile(f, JSONLHandler{})
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		if err := i.Synchronize(strings.NewReader("{\"test\":\"test1\"}\n")); err != nil {
+			t.Fatal(err)
+		}
+
+		r2 := strings.NewReader("{\"test\":\"test1\"}\n{\"test\":123}\n")
+		if err := i.Synchronize(r2); err != nil {
+			t.Fatal(err)
+		}
+
+		indexes, err := i.Indexes()
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		collected, err := indexes.Collect()
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		// check that the index file now has the additional index
+		if len(collected) != 2 {
+			t.Errorf("got len(i.Indexes) = %d, want 1", len(collected))
+		}
+
+		mp1, found, err := collected[0].BPTree(r2).Find([]byte("test1"))
+		if err != nil {
+			t.Fatal(err)
+		}
+		if !found {
+			t.Errorf("got i.Indexes[0].BPTree().Find(\"test1\") = nil, want non-nil")
+		}
+		if mp1.Offset != 0 || mp1.Length != uint32(len("{\"test\":\"test1\"}")) {
+			t.Errorf("got i.Indexes[0].BPTree().Find(\"test1\") = %+v, want {0, %d}", mp1, len("{\"test\":\"test1\"}"))
+		}
+
+		buf1, err := collected[0].Metadata()
+		if err != nil {
+			t.Fatal(err)
+		}
+		md1 := &appendable.IndexMeta{}
+		if err := md1.UnmarshalBinary(buf1); err != nil {
+			t.Fatal(err)
+		}
+		if md1.FieldType != appendable.FieldTypeString {
+			t.Errorf("got i.Indexes[0].FieldType = %#v, want FieldTypeString", md1.FieldType)
+		}
+
+		v2 := make([]byte, 8)
+		binary.LittleEndian.PutUint64(v2, math.Float64bits(123))
+		mp2, found, err := collected[1].BPTree(r2).Find(v2)
```

Comment:
It's unable to find `123` from the bptree. However, we can assert the `Find()` is not the issue as the `csv_test.go` has a test case that passes this assertion.

I don't see any glaring differences in the logic. The debugger indicates both values are in the bptree:

```sh
=== RUN   TestJSONL/existing_index_but_different_type
time=2024-02-02T10:28:43.780-05:00 level=DEBUG msg="handling json token" value=test1 start=8
time=2024-02-02T10:28:43.780-05:00 level=INFO msg="read line" i=0 offset=17
time=2024-02-02T10:28:43.780-05:00 level=DEBUG msg="handling json token" value=123 start=8
time=2024-02-02T10:28:43.780-05:00 level=INFO msg="read line" i=0 offset=30
```